File: sdks/python/examples/run_material_envelope.py
from __future__ import annotations

import json
import os

from kyuubiki_sdk import (
    ControlPlaneClient,
    KyuubikiAuth,
    material_study_envelope_catalog_request,
)


def main() -> None:
    base_url = os.getenv("KYUUBIKI_BASE_URL", "http://127.0.0.1:4000")
    token = os.getenv("KYUUBIKI_TOKEN")
    auth = KyuubikiAuth.access_token(token) if token else None

    client = ControlPlaneClient(base_url, auth=auth)
    request = material_study_envelope_catalog_request()
    job = client.submit_workflow_catalog_job(
        request["workflow_id"],
        request["input_artifacts"],
    )

    print(json.dumps(job, indent=2, sort_keys=True))


if __name__ == "__main__":
    main()

File: sdks/elixir/examples/run_material_envelope.exs
alias KyuubikiSdk.Auth
alias KyuubikiSdk.ControlPlaneClient

base_url = System.get_env("KYUUBIKI_BASE_URL", "http://127.0.0.1:4000")

auth =
  case System.get_env("KYUUBIKI_TOKEN") do
    nil -> nil
    token -> Auth.access_token(token)
  end

client = ControlPlaneClient.new(base_url, auth: auth)
request = KyuubikiSdk.material_study_envelope_catalog_request()

{:ok, job} =
  ControlPlaneClient.submit_workflow_catalog_job(
    client,
    request["workflow_id"],
    request["input_artifacts"]
  )

IO.puts(Jason.encode_to_iodata!(job, pretty: true))
